src/Controller/MediaAdminController.php

```diff
@@ -70,8 +72,12 @@ public function listAction(Request $request): Response
         $datagrid->setValue('context', null, $context);
 
         $rootCategory = null;
-        if ($this->has('sonata.media.manager.category')) {
-            $rootCategory = $this->get('sonata.media.manager.category')->getRootCategory($context);
+        if ($this->has('sonata.media.manager.category') && $this->has('sonata.media.manager.context')) {
+            $rootCategories = $this->get('sonata.media.manager.category')->getRootCategoriesForContext(
+                $this->get('sonata.media.manager.context')->find($context)
+            );
+
+            $rootCategory = $rootCategories[0];
```

Comment:
Same, `current($rootCategories)` is better I think

Reply:
Fixed.